code/modules/power/gravitygenerator.dm

```diff
@@ -382,22 +382,22 @@ GLOBAL_LIST_EMPTY(gravity_generators) // We will keep track of this by adding ne
 	return 0
 
 /obj/machinery/gravity_generator/main/proc/update_list()
-	var/list/levels_to_update = list()
-
-	if(ztrait)
-		for(var/level in SSmapping.levels_by_trait(ztrait))
-			levels_to_update += level
-	else if(!ztrait || !levels_to_update.len)
-		var/turf/T = get_turf(src.loc)
-		levels_to_update += T.z
-
-	for(var/level in levels_to_update)
-		if(!GLOB.gravity_generators["[level]"])
-			GLOB.gravity_generators["[level]"] = list()
-		if(on)
-			GLOB.gravity_generators["[level]"] |= src
+	var/turf/T = get_turf(src.loc)
+	if(T)
+		var/list/z_list = list()
+		// Multi-Z, station gravity generator generates gravity on all ZTRAIT_STATION z-levels.
+		if(SSmapping.level_trait(T.z, ZTRAIT_STATION))

```

Comment:
```suggestion
		if(ztrait)
```
I want to make a slight change so that we can have gravgens work on *anything* multi-z other than just the station.